### ISSUE TITLE
no resource plugin 'azure-v0.16.5-dev.1543258656'

### ISSUE THREAD

**Issue by dfinke** [pulumi/pulumi#2248]
`pulumi plugin ls` shows version `0.16.4` for `azure`.

I have pulumi 0.16.5 installed which solved the install issue for copying the binaries.

![image](https://user-images.githubusercontent.com/67258/49165005-3e8f6500-f2fe-11e8-84a0-d6b2c0665936.png)

**Comment by lukehoban** [comment]
@dfinke Curious - what version of `@pulumi/azure` are you referencing in your `package.json`?  Is it `dev`?

**Comment by lukehoban** [comment]
@dfinke Unless there's a specific reason you want that version, I'd suggest moving to `0.16.4` which is current latest.  `-dev` builds are nightlies, and `0.15.2` is ~3months old.

@swgillespie Any idea why we would be failing the resource provider installation here?

**Comment by dfinke** [comment]
I copied over the `cloud-ts-url-shortener-cache-http` example.

I changed to `"@pulumi/azure": "0.16.4"`, did an `npm install`, but still get the same error.

**Comment by lukehoban** [comment]
Can you share the output of your `npm install` and `pulumi plugin install resource azure v0.16.4`?  I just tried both and don't see any issues myself.  Also, which error exactly are you seeing still?

**Comment by swgillespie** [comment]
> @swgillespie Any idea why we would be failing the resource provider installation here?

Nope, I don't know why this would happen. In my mind either 1) `pulumi plugin install` is silently failing (a non-silent failure would fail `npm install` or 2) `pulumi plugin install` isn't being run but we're somehow changing the version of the package. The implication being that npm changed a package out from under us without running our post-install hook.

I don't know how we can get into a state where we think we need a plugin that doesn't exist either, which seems to be what @dfinke ran into.

**Comment by dfinke** [comment]
[log.txt](https://github.com/pulumi/pulumi/files/2625975/log.txt)

**Comment by swgillespie** [comment]
Thank you @dfinke!

Looks like the `0.16.5-dev` package is coming in through `@pulumi/cloud-azure`. That seems wrong, shouldn't we be shipping `@pulumi/cloud-` with dependencies on released versions?

**Comment by lukehoban** [comment]
> Looks like the 0.16.5-dev package is coming in through @pulumi/cloud-azure. That seems wrong, shouldn't we be shipping @pulumi/cloud- with dependencies on released versions?

Yep - that's a bug.  @CyrusNajmabadi or @ellismg we should fix this and do an update of `@pulumi/cloud-azure`.

**Comment by CyrusNajmabadi** [comment]
@lukehoban @ellismg  I've made the change to pulumi/cloud-azure.  Do you know what it takes to cut a release with that change in it?
